### PR TITLE
Fix containment and subslice optim. for steps.

### PR DESCRIPTION
**Pull request by anntzer**
#6497 set the `_path` attribute of Line2D to the actually drawn path
even if the drawstyle is `steps-*`; however containment tests and the
subslice optimization for very long paths were not updated accordingly
(see #6615).  This patch fixes the issues.

Note that `contains` returns, for events in a horizontal segment of a
"steps-mid" drawstyle plot, the index of the point in the segment,
regardless of whether the event occured to the left or the right of
that point.

### CODE REVIEW

File: lib/matplotlib/lines.py [hunk 1]
```diff
@@ -693,8 +696,9 @@ def _transform_path(self, subslice=None):
         """
         # Masked arrays are now handled by the Path class itself
         if subslice is not None:
-            _steps = self._path._interpolation_steps
-            _path = Path(self._xy[subslice, :], _interpolation_steps=_steps)
+            xy = STEP_LOOKUP_MAP[self._drawstyle](*self._xy[subslice, :].T)
+            _path = Path(np.asarray(xy).T,
```
Comment (tacaswell) [inline lib/matplotlib/lines.py]: Can you add a test image for this?

Comment (anntzer) [inline lib/matplotlib/lines.py]: done

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -4460,12 +4460,6 @@ def test_ls_ds_conflict():
                   linestyle='steps-pre:', drawstyle='steps-post')
 
 
-@cleanup
-def test_ls_ds_conflict():
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: why remove this?

Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: It is duplicated (see the lines just above).

Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: Ah, that is a good reason to remove it.

**Pull request merged**